### PR TITLE
Annotate exceptions while scanning perf output

### PR DESCRIPTION
**Pull request by cgaebel**
Include the line of perf output that caused scanf to raise in its error message.

### CODE REVIEW

File: src/perf_tool_backend.ml [hunk 1]
```diff
@@ -107,6 +107,7 @@ module Perf_line = struct
   let report_fields = "pid,tid,time,flags,addr,sym,symoff"
 
   let to_event line =
+    try
```
Comment (Xyene) [inline src/perf_tool_backend.ml:110]: Nit: indenting of the `try` body. https://github.com/janestreet/magic-trace/pull/30 should help.

**Review by Xyene — CHANGES_REQUESTED** [review]

File: src/perf_tool_backend.ml (hunk 1 above)
Comment (cgaebel) [reply src/perf_tool_backend.ml:110]: formatting passes now

File: src/perf_tool_backend.ml [hunk 2]
```diff
@@ -107,41 +107,49 @@ module Perf_line = struct
...
+                      ~unrecognized_perf_output:(line : string)])
+          ; addr
+          ; symbol
+          ; offset
+          })
+    with
+    | exn ->
+      raise_s
+        [%message
+          "exception raised while parsing perf output"
```
Comment (Xyene) [inline src/perf_tool_backend.ml:150]: "Please report this to ..."?

File: src/perf_tool_backend.ml [hunk 3]
```diff
@@ -107,41 +107,50 @@ module Perf_line = struct
...
+                      ~unrecognized_perf_output:(line : string)])
+          ; addr
+          ; symbol
+          ; offset
+          })
+    with
+    | exn ->
+      raise_s
+        [%message
+          "BUG: exception raised while parsing perf output Please report this to \
```
Comment (Xyene) [inline src/perf_tool_backend.ml:150]:
```suggestion
          "BUG: exception raised while parsing perf output. Please report this to \
```

**Review by Xyene — APPROVED** [review]
LGTM, thanks!

**Pull request merged**